#### Assemble and annotate a chloroplast genome

** What is genome assembly? **

* Genome assembly is the process of joining together DNA sequencing fragments into longer pieces, ideally up to chromosome lengths.
* The DNA fragments are called sequencing reads. These are in lengths of about 150 base pairs to up to a million (or more) base pairs, depending on the sequencing technology used. Currently, most reads are from Illumina (short), PacBio (long) or Oxford Nanopore (long and extra-long).
* It is difficult to assemble plant genomes as they are often large (for example, 3,000,000,000 base pairs), have many repeat regions (such as transposons), and may be polyploid.
* This tutorial shows genome assembly for a smaller data set - the plant chloroplast genome - a circular chromosome about 160,000 base pairs long.
* The process shown here is applicable to assembling plant nuclear genomes but there would be extra steps (and much more time) involved. For example, additional sequencing would usually be run on 10X, BioNano or HiC to produce information to link up longer nuclear chromosome pieces.

** Assemble a chloroplast genome **

* Go to: https://usegalaxy.org.au/
* This is Galaxy Australia: a free resource to use for genomics (and other) analyses.
* If you haven't used Galaxy before, see this tutorial [link] then return here.
* Log in.
